src/main/c/Jep/pyembed.c

```diff
@@ -703,6 +703,7 @@ void pyembed_thread_close(JNIEnv *env, intptr_t _jepThread)
     Py_DECREF(key);
 
     Py_CLEAR(jepThread->globals);
+    PyGC_Collect();
```

Comment:
Just curious, why do you the gc collect call before the Py_EndInterpreter?

Reply:
It's not specifically documented for PyGC_Collect but my default assumption for any Python c-api call is that it requires the thread to have an interpreter or at least hold the GIL and after Py_EndInterpreter the thread has neither.